sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -414,7 +415,7 @@ private[spark] object HiveUtils extends Logging {
         hadoopConf = hadoopConf,
         execJars = jars.toSeq,
         config = configurations,
-        isolationOn = true,
+        isolationOn = !isCliSessionState(),
```

Comment:
@yaooqinn Please take a look. `isCliSessionState` added by you.

Reply:
The isolation for non built-in hive shoud be on actually to isolate with those built in ones，right？

Reply:
sorry, I am unable to understand your mean, could you please elaborate me on this

Reply:
If you are using !built-in hive jars and turn off the isolation, which means using the SparkContxetClassLoader to load them directly , then you probably get hive class or  jar conflicts during the runtime

Reply:
okay I will recheck this part of the problem

Reply:
@wangyum if user is using  !built-in hive  then isolation should not be turned off, so this issue has to solved in different ways . 

1. Use in memory derby database. Configure hive-site.xml as below
```
<property>
<name>javax.jdo.option.ConnectionURL</name>
<value>jdbc:derby:memory:/opt/BigdataTools/spark-2.4.4-bin-hadoop2.7/metastore_db;create=true</value>
<description>JDBC connect string for a JDBC metastore</description>
</property>
```

2. Stop the derby instance started by [SparkSQLCLIDriver](https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala#L136) Something like below
![image](https://user-images.githubusercontent.com/35216143/66289178-793c3380-e8f9-11e9-99a9-4c1098918548.png)


@yaooqinn  what's your suggesstion

Reply:
IMO,  SparkSQLCLIDriver already load and initiate hive classes in main classloader, then we shouldn't use !builtin hive jars. Maybe just throw exception when using spark-sql stript with spark.sql.hive.metastore.jars

Reply:
there are 2 approach

1.based on `isCliSessionState`  Hive class should be loaded ( This PR fix)
2. Throw exception at the start of spark-sql if user used !builtin. By doing this there might be a problem, because currently this problem exists for Derby, if the user uses some other Database there will be a impact